app/services/exports/audience_export_service.rb

```diff
@@ -3,17 +3,42 @@
 require "csv"
 
 class Exports::AudienceExportService
-  def initialize(user)
+  FIELDS = ["Subscriber Email", "Subscribed Time"].freeze
+
+  def initialize(user, options = {})
     @user = user
+    @options = options.with_indifferent_access
+    timestamp = Time.current.to_fs(:db).gsub(/ |:/, "-")
+    @filename = "Subscribers-#{@user.username}_#{timestamp}.csv"
   end
 
+  attr_reader :filename, :tempfile
+
   def perform
-    CSV.generate do |csv|
-      csv << ["Follower Email", "Followed Time"]
+    @tempfile = Tempfile.new(["Subscribers", ".csv"], "tmp", encoding: "UTF-8")
+
+    CSV.open(@tempfile, "wb") do |csv|
+      csv << FIELDS
```

Comment:
nit: could we move this to the `.open(..., headers: [ ... ], write_headers: true)`